### PR TITLE
Commenting modes app projection algorithm

### PR DESCRIPTION
**Pull request by bhourahine**
Comments on the projection algorithm in modes.

### CODE REVIEW

File: app/modes/modeprojection.F90 [hunk 1]
```diff
@@ -50,36 +50,40 @@ subroutine project(dynMatrix, tRemoveTranslate, tRemoveRotate, nDerivs, nMovedAt
...
-        do ii = 1, 3
-          vectorsToNull((iAt - 1) * 3 + ii, ii) = 1.0_dp
-        end do
-      end do
+       ! Vectors for each atom to point along each of the Cartesian directions
+       do iAt = 1, nMovedAtom
+          do ii = 1, 3
+             vectorsToNull((iAt - 1) * 3 + ii, ii) = 1.0_dp
+          end do
+       end do
```
Comment (vanderhe) [inline app/modes/modeprojection.F90]: @bhourahine Indentation off?

Comment (bhourahine) [reply app/modes/modeprojection.F90]: Fixed the examples in this file, but I _think_ there might be some problems with the default .emacs file for very recent emacs.

Comment (bhourahine) [reply app/modes/modeprojection.F90]: @vanderhe As this is a PR into your branch, I can't merge it when approved.

Comment (vanderhe) [reply app/modes/modeprojection.F90]: Who uses stock .emacs? :wink:

**Review by vanderhe — APPROVED** [review]
Thx!

File: app/modes/modeprojection.F90 (hunk 1 above)
Comment (bhourahine) [reply app/modes/modeprojection.F90]: I mean _our_ [.emacs](https://dftbplus-develguide.readthedocs.io/en/latest/languages/fortranstyle.html), ws-butler is also throwing warnings with emacs 29.3.

Comment (vanderhe) [reply app/modes/modeprojection.F90]: Seems to be 'slightly' outdated. With the new project.el functionality Emacs is better than ever, even no need to resort to projectile anymore. Unfortunately Bálint betrayed Emacs and defected to Microsoft, sad but true... :sweat_smile:

Comment (bhourahine) [reply app/modes/modeprojection.F90]: Meaning the https://github.com/dftbplus/develguide Fortran section about editors needs updating.

**Pull request merged**